pep-0011.txt

```diff
@@ -14,109 +14,137 @@ Post-History: 18-Aug-2007,
-
-CPU architecture and compiler support are viewed in a similar manner
-as platforms. For example, to consider the ARM architecture supported
-a buildbot running on ARM would be required along with support from
-the Python development team. In general it is not required to have
-a CPU architecture run under every possible platform in order to be
-considered supported.
-
-Unsupporting platforms
-----------------------
-
-If a certain platform that currently has special code in CPython is
-deemed to be without enough Python users or lacks proper support from
-the Python development team and/or a buildbot, a note must be posted
-in this PEP that this platform is no longer actively supported.  This
-note must include:
-
-- the name of the system
-- the first release number that does not support this platform
-  anymore, and
-- the first release where the historical support code is actively
-  removed
-
-In some cases, it is not possible to identify the specific list of
-systems for which some code is used (e.g. when autoconf tests for
-absence of some feature which is considered present on all
-supported systems).  In this case, the name will give the precise
-condition (usually a preprocessor symbol) that will become
-unsupported.
-
-At the same time, the CPython source code must be changed to
-produce a build-time error if somebody tries to install Python on
-this platform.  On platforms using autoconf, configure must fail.
-This gives potential users of the platform a chance to step
-forward and offer maintenance.
-
-
-Re-supporting platforms
------------------------
-
-If a user of a platform wants to see this platform supported
-again, they may volunteer to maintain the platform support.  Such an
-offer must be recorded in the PEP, and the user can submit patches
-to remove the build-time errors, and perform any other maintenance
-work for the platform.
+This PEP also lists what plaforms *are* supported by the CPython
+interpreter. This lets people know what platforms are directly
+supported by the CPython development team.
+
+
+Support tiers
+=============
+
+Platform support is broken down into *tiers*. Each tier comes with
+different requirements which lead to different promises being made
+about support.
+
+To be promoted to a tier, steering council support is required and is
+expected to be driven by team consensus. Demotion to a lower tier
+occurs then the requirements of the current tier are no longer met for
+a platform for an extended period of time based on the judgment of
+the release manager or steering council. For platforms which no longer
+meet the requirements of any tier by b1 of a new feature release, an
+announcement will be made to warn the community of the pending removal
+of support for the platform (e.g. in the b1 announcement). If the
+platform is not brought into line for at least one of the tiers by the
+first release candidate, it will be listed as unsupported in this PEP.
```

Comment:
```suggestion
Platform support is broken down into *tiers*. Each tier comes with
different requirements which lead to different promises being made
about support.

To be promoted to a tier, steering council support is required and is
expected to be driven by team consensus. Demotion to a lower tier
occurs when the requirements of the current tier are no longer met for
a platform for an extended period of time based on the judgment of
the release manager or steering council. For platforms that no longer
meet the requirements of any tier by b1 of a new feature release, an
announcement will be made to warn the community of the pending removal
of support for the platform (e.g. in the b1 announcement). If the
platform is not brought into line for at least one of the tiers by the
first release candidate, it will be listed as unsupported in this PEP.
```